### ISSUE TITLE
get_or_create methods for high-level api

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos-cuber#196]
As mentioned in https://github.com/scalableminds/webknossos-cuber/pull/165#issuecomment-565017245, it would be useful to have an `get_or_create` interface when dealing with datasets, layers and mags. Behavior in pseudo code should be:

```
def get_or_create(params):
  if exists:
    assert dataset.params == params
  else:
    create(params)
```

**Comment by rschwanhold** [comment]
There are such methods for layers and mags:
https://github.com/scalableminds/webknossos-cuber/blob/f912ff5719447633526bbf24ff2f6049a342f1b8/wkcuber/api/Dataset.py#L79

https://github.com/scalableminds/webknossos-cuber/blob/f912ff5719447633526bbf24ff2f6049a342f1b8/wkcuber/api/Layer.py#L73
(The method `get_or_add_mag()` is defined in the subclasses. Maybe that is why you didn't see it)

However, there is no such method to get or create a `Dataset`.

**Comment by philippotto** [comment]
Ah thanks, very helpful and good to know! Then, it's only missing for `Dataset`.

**Comment by rschwanhold** [comment]
merged in #220
